src/track.c

```diff
@@ -676,12 +675,18 @@ void calc_navigation_measurement_(u8 n_channels, channel_measurement_t* meas[],
     nav_meas[i]->carrier_phase += (nav_time - meas[i]->receiver_time) * meas[i]->carrier_freq;
 
     nav_meas[i]->lock_counter = meas[i]->lock_counter;
+	
+	// calc the clock error
+	calc_sat_state(ephemerides[i], nav_meas[i]->tot,
+                   nav_meas[i]->sat_pos, nav_meas[i]->sat_vel,
+                   &clock_err, &clock_rate_err);
```

Comment:
Also, I believe the satellite clock error is small enough (<1% of pseudorange) that this effect isn't very significant, but if its possible to do it this way without additional overhead then it sounds good to me.